models/refresh_token.go

```diff
@@ -70,9 +70,22 @@ func RevokeTokenFamily(tx *storage.Connection, token *RefreshToken) error {
 	return nil
 }
 
-// Logout deletes all refresh tokens for a user.
-func Logout(tx *storage.Connection, instanceID uuid.UUID, id uuid.UUID) error {
-	return tx.RawQuery("DELETE FROM "+(&pop.Model{Value: RefreshToken{}}).TableName()+" WHERE instance_id = ? AND user_id = ?", instanceID, id).Exec()
+// Logout either deletes a specific refresh token and its descendants, or deletes all refresh tokens for a user.
+func Logout(tx *storage.Connection, instanceID uuid.UUID, id uuid.UUID, refreshToken string) error {
+	if refreshToken == "" {
+		// Delete all refresh tokens.
+		return tx.RawQuery("DELETE FROM "+(&pop.Model{Value: RefreshToken{}}).TableName()+" WHERE instance_id = ? AND user_id = ?", instanceID, id).Exec()
+	} else {
+		// Delete the given refresh token, and all its descendants.
```

Comment:
I'm guesssing I should also delete ancestor refresh tokens here?

Reply:
yeah if we make the assumption that a refresh token family == a session, we should delete all ancestor and children refresh tokens